### PR TITLE
Add support for multicast in user defined networks.

### PR DESCRIPTION
**Pull request by dceara**
#### What this PR does and why is it needed
This adds:
- cluster and router port groups to user defined L2/L3 primary networks
- multicast support to user defined L2/L3 primary networks
- hairpin ACLs allowing hairpin service traffic for user defined L2/L3 primary networks too

#### Special notes for reviewers

Cluster-wide port group management routines have been moved to the base network controller and now are used by secondary network controllers too.

Multicast ACL management routines have been moved to the base network controller and now are used by secondary network controllers too.

E2e and unit tests have been adapted to run for both default and primary UDN networks for the features mentioned above.

#### How to verify it

Unit and end to end tests have been adapted/created.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go [hunk 1]
```diff
@@ -336,9 +336,9 @@ func NewSecondaryLayer2NetworkController(cnci *CommonNetworkControllerInfo, netI
 			claimsReconciler)
 	}
 
-	// disable multicast support for secondary networks
-	// TBD: changes needs to be made to support multicast in secondary networks
-	oc.multicastSupport = false
+	// enable multicast suppor at UDN only for primaries + multicast enabled
```
Comment (trozet) [inline go-controller/pkg/ovn/secondary_layer2_network_controller.go:339]: nit: support

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go [hunk 2]
```diff
@@ -336,9 +336,9 @@ func NewSecondaryLayer2NetworkController(cnci *CommonNetworkControllerInfo, netI
...
+	// TBD: changes needs to be made to support multicast beyond primary UDN
+	oc.multicastSupport = util.IsNetworkSegmentationSupportEnabled() && config.EnableMulticast
```
Comment (trozet) [inline go-controller/pkg/ovn/secondary_layer2_network_controller.go:341]: shouldn't you also check if this controller is serving a primary network?

File: go-controller/pkg/ovn/secondary_layer3_network_controller.go [hunk 3]
```diff
@@ -376,9 +376,9 @@ func NewSecondaryLayer3NetworkController(cnci *CommonNetworkControllerInfo, netI
 		oc.podAnnotationAllocator = podAnnotationAllocator
 	}
 
-	// disable multicast support for secondary networks
-	// TBD: changes needs to be made to support multicast in secondary networks
-	oc.multicastSupport = false
+	// enable multicast suppor at UDN only for primaries + multicast enabled
+	// TBD: changes needs to be made to support multicast beyond primary UDN
+	oc.multicastSupport = util.IsNetworkSegmentationSupportEnabled() && config.EnableMulticast
```
Comment (trozet) [inline go-controller/pkg/ovn/secondary_layer3_network_controller.go:381]: here as well for primary check

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go [hunk 4]
```diff
@@ -432,11 +432,15 @@ func (oc *SecondaryLayer2NetworkController) Init() error {
 		return err
 	}
 
-	// Configure cluster port groups for user defined primary networks.
+	// Configure cluster port groups abd multicast default policies for user defined primary networks.
```
Comment (trozet) [inline go-controller/pkg/ovn/secondary_layer2_network_controller.go:435]: s/abd/and

**Review by trozet — COMMENTED** [review]
nice work @qinqon and @dceara. Very clean and easy to follow. Just really 1 minor comment about checking if network controller is serving a primary role. Probably should also rebase.

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go (hunk 1 above)
Comment (dceara) [reply go-controller/pkg/ovn/secondary_layer2_network_controller.go]: ack, fixed

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go (hunk 2 above)
Comment (dceara) [reply go-controller/pkg/ovn/secondary_layer2_network_controller.go]: ack, fixed

File: go-controller/pkg/ovn/secondary_layer2_network_controller.go (hunk 4 above)
Comment (dceara) [reply go-controller/pkg/ovn/secondary_layer2_network_controller.go]: ack, fixed

File: go-controller/pkg/ovn/secondary_layer3_network_controller.go (hunk 3 above)
Comment (dceara) [reply go-controller/pkg/ovn/secondary_layer3_network_controller.go]: ack, fixed

**Review by trozet — APPROVED** [review]

**Pull request merged**